### PR TITLE
Update cairo vm version.

### PR DESCRIPTION
**Pull request by alonh5**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)

### CODE REVIEW

**Review by elintul — CHANGES_REQUESTED** [review]
> Reviewed 16 of 16 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: all files reviewed, 3 unresolved discussions (waiting on @alonh5, @dorimedini-starkware, @noaov1, and @spapinistarkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQMX5365t_Dvb-Rsas:-NXQMX5365t_Dvb-Rsat:b-l9ujoc) (no related file):*
Yay, you found my new methods!
Can you make sure they are used throughout the code, so that we will have only one location to change with any future change in the VM program getters? 

___
___
*[`crates/blockifier/src/execution/cairo1_execution.rs` line 217 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQL_BC-B0TVCmP9wO6:-NXQL_BC-B0TVCmP9wO7:b9v2kp0) ([raw file](https://github.com/starkware-libs/blockifier/blob/7c63d2e36b76eb0ecc460281f93a546a34513fa3/crates/blockifier/src/execution/cairo1_execution.rs#L217)):*
> ```XML
>         entry_point.pc(),
>         &args,
>         &mut None,
> ```

Give the new argument a name? It's a bit cryptic like that. 
___
*[`crates/blockifier/src/execution/deprecated_execution.rs` line 194 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQMIbz0E2I4SLlx1MN:-NXQMIbz0E2I4SLlx1MO:bgy6baz) ([raw file](https://github.com/starkware-libs/blockifier/blob/7c63d2e36b76eb0ecc460281f93a546a34513fa3/crates/blockifier/src/execution/deprecated_execution.rs#L194)):*
> ```XML
>         &args,
>         &mut None,
>         verify_secure,
> ```

Here too.

**Review by elintul — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: 14 of 16 files reviewed, 3 unresolved discussions (waiting on @alonh5, @dorimedini-starkware, @noaov1, and @spapinistarkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQMX5365t_Dvb-Rsas:-NXQSp5q8sZqm3_s4A6k:b2nh4fg) (no related file):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Luckily, they made the fields previously used private, so the compiler made sure for me :)
</blockquote></details>

:) 

___
___
*[`crates/blockifier/src/execution/cairo1_execution.rs` line 217 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQL_BC-B0TVCmP9wO6:-NXQS1fs8wAv-NwfHHmW:b-j9zz5t) ([raw file](https://github.com/starkware-libs/blockifier/blob/7c63d2e36b76eb0ecc460281f93a546a34513fa3/crates/blockifier/src/execution/cairo1_execution.rs#L217)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Why isn't it needed?

**Review by elintul — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583#-:-NXQTKiG4D3xSB89bebM:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @alonh5, @dorimedini-starkware, @noaov1, and @spapinistarkware)

**Review by elintul — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: all files reviewed, 4 unresolved discussions (waiting on @alonh5, @noaov1, and @spapinistarkware)

**Review by alonh5 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: 14 of 16 files reviewed, 3 unresolved discussions (waiting on @dorimedini-starkware, @elintul, @noaov1, and @spapinistarkware)

___
*[`crates/blockifier/src/test_utils.rs` line 342 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQa4BW5yt-9hl_IHjR:-NXaD0Jv1EsC7JTLrb7j:b-jfm05q) ([raw file](https://github.com/starkware-libs/blockifier/blob/f03ad714e0215994f0310333559c47423181a9d1/crates/blockifier/src/test_utils.rs#L342)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

why `String::from` in one place and `to_string()` everywhere else...? non blocking, just strange
</blockquote></details>

Changed it.
___
*[`crates/blockifier/src/execution/cairo1_execution.rs` line 217 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQL_BC-B0TVCmP9wO6:-NXQk_JuB5FTeXFegG_V:b-gavpr9) ([raw file](https://github.com/starkware-libs/blockifier/blob/7c63d2e36b76eb0ecc460281f93a546a34513fa3/crates/blockifier/src/execution/cairo1_execution.rs#L217)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Why isn't it needed? 
</blockquote></details>

WDYM? I changed as you asked.
___
*[`crates/blockifier/src/fee/fee_test.rs` line 33 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQbW1_2dQvSKsxvm5B:-NXaDK9E4iVJhra7D8Xv:b-896fix) ([raw file](https://github.com/starkware-libs/blockifier/blob/f03ad714e0215994f0310333559c47423181a9d1/crates/blockifier/src/fee/fee_test.rs#L33)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

replace with constant
</blockquote></details>

Done.
___
*[`crates/blockifier/src/fee/os_resources.rs` line 11 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQb_s25thiVJkHhGyq:-NXaDw3KCc8PJSzT29Cv:bps7m12) ([raw file](https://github.com/starkware-libs/blockifier/blob/f03ad714e0215994f0310333559c47423181a9d1/crates/blockifier/src/fee/os_resources.rs#L11)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

why not use constants here for all builtin names + n_steps, like I did?
</blockquote></details>

Generally, this file is autogenerated in the Python repo and copied here, so it doesn't add anything to use the constants because it will be overwritten anyway. Eventually, we will probably want to change the names in the autogeneration when it is only used here in blockifer.

**Review by elintul — DISMISSED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: all files reviewed, 3 unresolved discussions (waiting on @alonh5, @dorimedini-starkware, @noaov1, and @spapinistarkware)

___
*[`crates/blockifier/src/test_utils.rs` line 342 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQa4BW5yt-9hl_IHjR:-NXe1YXb66dMwLkB4UG1:b-kq5pks) ([raw file](https://github.com/starkware-libs/blockifier/blob/f03ad714e0215994f0310333559c47423181a9d1/crates/blockifier/src/test_utils.rs#L342)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Changed it.
</blockquote></details>

Briefly go through the codebase and make sure it's consistent?
___
*[`crates/blockifier/src/execution/cairo1_execution.rs` line 217 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQL_BC-B0TVCmP9wO6:-NXe1VX32Az4t39AmsiO:b4rziiw) ([raw file](https://github.com/starkware-libs/blockifier/blob/7c63d2e36b76eb0ecc460281f93a546a34513fa3/crates/blockifier/src/execution/cairo1_execution.rs#L217)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

WDYM? I changed as you asked.
</blockquote></details>

Why isn't this argument used? Why do we pass `None`?
___
*[`crates/blockifier/src/fee/os_resources.rs` line 11 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQb_s25thiVJkHhGyq:-NXe1psT-rUy1mWV0MpG:b-v3gvj1) ([raw file](https://github.com/starkware-libs/blockifier/blob/f03ad714e0215994f0310333559c47423181a9d1/crates/blockifier/src/fee/os_resources.rs#L11)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Generally, this file is autogenerated in the Python repo and copied here, so it doesn't add anything to use the constants because it will be overwritten anyway. Eventually, we will probably want to change the names in the autogeneration when it is only used here in blockifer.
</blockquote></details>

"like I did?"
Where?

**Review by dorimedini-starkware — COMMENTED** [review]
> Reviewed 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: 14 of 16 files reviewed, all discussions resolved (waiting on @elintul, @noaov1, and @spapinistarkware)

___
*[`crates/blockifier/src/fee/os_resources.rs` line 11 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQb_s25thiVJkHhGyq:-NXe1vHHExHo9foHkp3v:b-6lb8tk) ([raw file](https://github.com/starkware-libs/blockifier/blob/f03ad714e0215994f0310333559c47423181a9d1/crates/blockifier/src/fee/os_resources.rs#L11)):*
> "like I did?"

in my WIP PR where I tried updating LC to a non-tagged version.

> it will be overwritten anyway

I think you're right, [PTAL at this name consistency test PR](https://reviewable.io/reviews/starkware-libs/blockifier/587)

**Review by dorimedini-starkware — COMMENTED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: all files reviewed, 1 unresolved discussion (waiting on @alonh5, @noaov1, and @spapinistarkware)

___
*[`Cargo.toml` line 13 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXe2NMuCGiUYVNtwhZp:-NXe2NMuCGiUYVNtwhZq:b-328xya) ([raw file](https://github.com/starkware-libs/blockifier/blob/f883656acfd57969452f01b8f41c379cd3e6bef0/Cargo.toml#L13)):*
> ```TOML
> cairo-lang-casm = { git = "https://github.com/starkware-libs/cairo", rev = "e71724e4" }
> cairo-lang-runner = { git = "https://github.com/starkware-libs/cairo", rev = "e71724e4" }
> cairo-lang-starknet = { git = "https://github.com/starkware-libs/cairo", rev = "e71724e4" }
> ```

ah... maybe ask them to tag a commit...?
will need papyrus to tag something as well...
is using these revisions problematic?



_Code quote:_
```TOML
cairo-lang-casm = { git = "https://github.com/starkware-libs/cairo", rev = "e71724e4" }
cairo-lang-runner = { git = "https://github.com/starkware-libs/cairo", rev = "e71724e4" }
cairo-lang-starknet = { git = "https://github.com/starkware-libs/cairo", rev = "e71724e4" }
```

**Review by elintul — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: all files reviewed, 2 unresolved discussions (waiting on @alonh5, @noaov1, and @spapinistarkware)

___
*[`Cargo.toml` line 24 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXe2WjRDu-em0dlOTU_:-NXe2WjRDu-em0dlOTUa:blaliqm) ([raw file](https://github.com/starkware-libs/blockifier/blob/f883656acfd57969452f01b8f41c379cd3e6bef0/Cargo.toml#L24)):*
> ```TOML
> num-traits = "0.2"
> ouroboros = "0.15.6"
> papyrus_storage = { git = "https://github.com/starkware-libs/papyrus", rev = "e71a1fc" }
> ```

Was is updated again?

**Review by elintul — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: all files reviewed, 2 unresolved discussions (waiting on @alonh5, @noaov1, and @spapinistarkware)

**Review by alonh5 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: all files reviewed, 2 unresolved discussions (waiting on @dorimedini-starkware, @elintul, @noaov1, and @spapinistarkware)

___
*[`Cargo.toml` line 13 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXe2NMuCGiUYVNtwhZp:-NXe5Ehf5SZy55isIRA6:b-s21lev) ([raw file](https://github.com/starkware-libs/blockifier/blob/f883656acfd57969452f01b8f41c379cd3e6bef0/Cargo.toml#L13)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

ah... maybe ask them to tag a commit...?
will need papyrus to tag something as well...
is using these revisions problematic?
</blockquote></details>

I don't think it's problematic. They're releasing a new version soon, but I wouldn't want to wait for that (note we will have to do this update sequence any time we want to get a newer version of cairo-vm).
___
*[`Cargo.toml` line 24 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXe2WjRDu-em0dlOTU_:-NXe3b1P-f-uc6pt_OX2:bl3iy17) ([raw file](https://github.com/starkware-libs/blockifier/blob/f883656acfd57969452f01b8f41c379cd3e6bef0/Cargo.toml#L24)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Was is updated again?
</blockquote></details>

Yes, Dan added something (you can see the thread about it in #full-node-sequencer).
___
*[`crates/blockifier/src/test_utils.rs` line 342 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQa4BW5yt-9hl_IHjR:-NXe4GOU560sAKgNZaO7:bpvhpqi) ([raw file](https://github.com/starkware-libs/blockifier/blob/f03ad714e0215994f0310333559c47423181a9d1/crates/blockifier/src/test_utils.rs#L342)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Briefly go through the codebase and make sure it's consistent?
</blockquote></details>

Only `.to_string()` is used for the builtin names but `String::from()` *is* used in some other places. Do we want to only use one of them in the whole codebase?
___
*[`crates/blockifier/src/execution/cairo1_execution.rs` line 217 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQL_BC-B0TVCmP9wO6:-NXe28Wi2gwcp0lVFkMI:bckrbgl) ([raw file](https://github.com/starkware-libs/blockifier/blob/7c63d2e36b76eb0ecc460281f93a546a34513fa3/crates/blockifier/src/execution/cairo1_execution.rs#L217)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

Why isn't this argument used? Why do we pass `None`?
</blockquote></details>

Oh, IIUC it's used to limit the resources in the execution (i.e., stopping when the wrapped "n_steps" have been done).
We are going to want to use it, right? It can be done in a subsequent PR.

**Review by dorimedini-starkware — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: all files reviewed, 1 unresolved discussion (waiting on @elintul, @noaov1, and @spapinistarkware)

___
*[`Cargo.toml` line 13 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXe2NMuCGiUYVNtwhZp:-NXe6gup2Fe11uU08XhD:bth9mle) ([raw file](https://github.com/starkware-libs/blockifier/blob/f883656acfd57969452f01b8f41c379cd3e6bef0/Cargo.toml#L13)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

I don't think it's problematic. They're releasing a new version soon, but I wouldn't want to wait for that (note we will have to do this update sequence any time we want to get a newer version of cairo-vm).
</blockquote></details>

It's a problem if you also have local dependencies, but as long as we don't have them... no problem.

**Review by elintul — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: all files reviewed, 1 unresolved discussion (waiting on @alonh5, @noaov1, and @spapinistarkware)

___
*[`Cargo.toml` line 24 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXe2WjRDu-em0dlOTU_:-NXeBf8JCYjBheptBJ8Z:bnjzhup) ([raw file](https://github.com/starkware-libs/blockifier/blob/f883656acfd57969452f01b8f41c379cd3e6bef0/Cargo.toml#L24)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Yes, Dan added something (you can see the thread about it in #full-node-sequencer).
</blockquote></details>

TL;DR? Why wasn't it needed before?
___
*[`crates/blockifier/src/test_utils.rs` line 342 at r3](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQa4BW5yt-9hl_IHjR:-NXeBWs2Dl6VaS9MIBEb:bb28d17) ([raw file](https://github.com/starkware-libs/blockifier/blob/f03ad714e0215994f0310333559c47423181a9d1/crates/blockifier/src/test_utils.rs#L342)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Only `.to_string()` is used for the builtin names but `String::from()` *is* used in some other places. Do we want to only use one of them in the whole codebase?
</blockquote></details>

Depends on the case, perhaps not in this PR.
___
*[`crates/blockifier/src/execution/cairo1_execution.rs` line 217 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQL_BC-B0TVCmP9wO6:-NXeBaazBhjImpyyXN62:b-bm1i2r) ([raw file](https://github.com/starkware-libs/blockifier/blob/7c63d2e36b76eb0ecc460281f93a546a34513fa3/crates/blockifier/src/execution/cairo1_execution.rs#L217)):*
<details><summary><i>Previously, alonh5 (Alon Haramati) wrote…</i></summary><blockquote>

Oh, IIUC it's used to limit the resources in the execution (i.e., stopping when the wrapped "n_steps" have been done).
We are going to want to use it, right? It can be done in a subsequent PR.

</blockquote></details>

From what I understood, LC removed this functionality.

**Review by elintul — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @noaov1 and @spapinistarkware)

**Review by alonh5 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/583)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @noaov1 and @spapinistarkware)

___
*[`Cargo.toml` line 24 at r5](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXe2WjRDu-em0dlOTU_:-NXeDlpx3OzP8F0z_x1k:bg1cx2o) ([raw file](https://github.com/starkware-libs/blockifier/blob/f883656acfd57969452f01b8f41c379cd3e6bef0/Cargo.toml#L24)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

TL;DR? Why wasn't it needed before?
</blockquote></details>

He removed something from papyrus because it was previously moved to SN API. It wasn't necessary just thought I'd take the latest revision.
___
*[`crates/blockifier/src/execution/cairo1_execution.rs` line 217 at r1](https://reviewable.io/reviews/starkware-libs/blockifier/583#-NXQL_BC-B0TVCmP9wO6:-NXeE8OR6oagIhjSNSXv:bfuqvkx) ([raw file](https://github.com/starkware-libs/blockifier/blob/7c63d2e36b76eb0ecc460281f93a546a34513fa3/crates/blockifier/src/execution/cairo1_execution.rs#L217)):*
<details><summary><i>Previously, elintul (Elin) wrote…</i></summary><blockquote>

From what I understood, LC removed this functionality.
</blockquote></details>

Removed it? Didn't they just add it?
In any case, it's included in the version they released.
